pkg/gitops/github.go

```diff
@@ -0,0 +1,114 @@
+package gitops
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/bitrise-io/go-steputils/stepconf"
+	gogh "github.com/google/go-github/v33/github"
+	"golang.org/x/oauth2"
+)
+
+//go:generate moq -out github_moq_test.go . githuber
+type githuber interface {
+	AddKey(context.Context, []byte) (int64, error)
+	DeleteKey(context.Context, int64) error
+	OpenPullRequest(context.Context, openPullRequestParams) (string, error)
+}
+
+// github implements the githuber interface.
+var _ githuber = (*github)(nil)
```

Comment:
I wanted to make sure that `github` implements the `githuber` interface even if it's not used that way in the code (so even if it's not passed into a `githuber` interface anywhere in this repositories code).

So I'm ensuring interface implementation explicitly here instead of go's implicit (duck typing) interface implementation, getting a fast feedback loop at this point if the contract is broken. I found this a good practice in some cases so far (it's also suggested in most style guides, ie [Uber's](https://github.com/uber-go/guide/blob/master/style.md#verify-interface-compliance))

Reply:
It seems a bit weird for a not exposed interface/type, but at least the methods are exported, so it can be regarded as part of the contract